proposals/load_result.json

```diff
@@ -1,7 +1,7 @@
 {
     "id": "load_result",
     "summary": "Load batch job results",
-    "description": "Loads batch job results by job id from the server-side user workspace. The job must have been stored by the authenticated user on the back-end currently connected to.",
+    "description": "Loads batch job results and returns them as a processable data cube. A batch job result can be loaded by ID or URL:\n\n* **ID**: A batch job identifier available on the server-side user workspace. The job must have been stored by the authenticated user on the back-end currently connected to.\n* **URL**: The URL to the STAC metadata for a batch job result. This needs to the signed URL that is provided by some back-ends through the `canonical` link relation in the batch job result metadata. Please note that this is only supported since openEO API version 1.1.0.\n\nIf supported by the underlying metadata and file format, the data that is added to the data cube can be restricted with the parameters `spatial_extent`, `temporal_extent` and `bands`.\n\n**Remarks:**\n\n* The bands (and all dimensions that specify nominal dimension labels) are expected to be ordered as specified in the metadata if the `bands` parameter is set to `null`.\n* If no additional parameter is specified this would imply that the whole data set is expected to be loaded. Due to the large size of many data sets, this is not recommended and may be optimized by back-ends to only load the data that is actually required after evaluating subsequent processes such as filters. This means that the pixel values should be processed only after the data has been limited to the required extent and as a consequence also to a manageable size.",
```

Comment:
> and then you are screwed without signed URLs.

I wasn't saying to not support signed URLs, just questioning if the description has to specify that it _must_ be a signed URL.

FYI: there is a verb missing here as well:

> This needs to the signed URL that ...

Reply:
So you are proposing that we should allow also "non-result" to be importable, e.g. a "foreign" STAC catalog?

I'll fix the missing verb, thanks.

Reply:
Anyway, does this sound better?

> This is usually a signed URL that is provided by some back-ends since openEO API version 1.1.0 through the `canonical` link relation in the batch job result metadata.

Reply:
yes, I like that "This is usually a signed URL"  describes the main approach, but leaves some wiggle room for alternative implementations

Reply:
So I think this thread is resolved, right? Did I miss anything?

Reply:
Another remark about this last sentence about result filtering:

> This means that the pixel values should be processed only after the data has been limited to the required extent and as a consequence also to a manageable size.

As already noted above and elsewhere: I think that something like "process pixels after limiting" is an implementation detail that the user should have not worry about on the level of openEO process graphs. For example, a database user that writes a SQL query does not have to think about the order of actual data processing steps related to  SELECT/FROM/WHERE/... operations, the user just expects the SQL engine to do the smart/right thing.

Moreover, if you want to run a secondary service on your batch job result: you usually don't want to hard-code spatial/temporal filters in your service's process graph. These filters will be append by the secondary service "at run time" at the end the actual process graph, so there is  a bit of conflict with "pixel values should be processed only after the data has been limited".

I would just propose to drop the sentence "This means that the pixel values ... to a manageable size". I'm fine with keeping the preceding sentences

Reply:
Hmm, I think it's useful to say that a user should not load the whole world, then do various processing steps and just at the end limit the extent for example after a lot of CPU hours could've been wasted. In SQL you also need to follow some principles as otherwise it might get costly (e.g. create a filter if you filter on the column a lot). The sentence you are referring to is meant to help the user structure the process graph, this is *not* meant to be a hint for the back-end implementer. But maybe this needs to be rewritten (also in load_collection then).

With regards to the web services: Depending on the web service, it can expose variables that you can pass to load_collection or other processes so that you don't need to hardcode these.

Reply:
Is this better?

> If no additional parameter is specified this implies that the whole data set is expected to be loaded. Back-ends may optimize this, but it is still recommended that users limit their data cubes to the required size as soon as possible.

Reply:
fine-tuning suggestion:

> Not specifying any additional parameter implies to load the whole data set, which might be expensive performance-wise. Back-ends may optimize this with lazy loading techniques, but it is generally recommended to limit the loaded data as soon as possible to the relevant spatio-temporal extent.

Reply:
> expensive performance-wise

It might be expensive in different regards, not sure I'd specifically mention performance here.

> lazy loading techniques

Is this detail important? I thought we are going to be less technical. ;-)

> to the relevant spatio-temporal extent.

This doesn't capture the bands (and in load_collection: properties).

A merge of both?

> Not specifying any additional parameter implies loading the whole data set into a data cube, which might be expensive. Back-ends may optimize this, but it is generally recommended to limit the data cubes to the required size as soon as possible.

Reply:
A bare "expensive" sounds money-related to me. In the end it will be reflected in the billing, but I guess users should be worried first about performance or computational cost.

"lazy loading" is indeed implementation detail. Fine to omit.

"required size": I wanted to avoid "size", because that is not exposed explicitly, let alone controllable by the user, they control spatio-temporal extents. The fact that the description doesn't cover bands or properties is more something "left as exercise to the reader".  Possibile alternative: "data subset".

Reply:
expensive: I added performance/cost.

size: I've checked the xarray docs as an example and they seem to also call it "size", so I'd expect that a good amount of users understand it.

Reply:
It's  just that "the required size"  is confusing to me  as a simple user: I have no idea about the actual size (is it in pixels, tiles, file bytes, memory bytes?)  and I don't know what is "required" (is it required for my use case or is it a back-end limitation?). To play the devil's advocate: a naive user might also read the "size" recommendation  as "I can load the whole world, if I down-sample it early enough". If you talk about extents, you don't have that interpretation.

proposal:
> .. to limit the data cube to the relevant extent as soon as possible.